components/ip2whois/actions/query-domain-info/query-domain-info.mjs

```diff
@@ -0,0 +1,35 @@
+import ip2whoisApp from "../../ip2whois.app.mjs";
+
+export default {
+  name: "WHOIS lookup",
+  description: "Helps users to obtain domain information, WHOIS record, by using a domain name. Please refer to the [documentation](https://ip2whois.com/developers-api) for the details of the fields returned.",
+  key: "ip2whois-query-domain-info",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    ip2whoisApp,
+    domain: {
+      type: "string",
+      label: "Domain name",
+      description: "Domain name.",
+    },
+    format: {
+      type: "string",
+      label: "Response Format",
+      description: "Format of the response message. Available values are `json` or `xml`. If unspecified, json format will be used for the response message.",

```

Comment:
You should add an `options` prop to make this easier for the user. For example:

```
      options: [
        "JSON",
        "XML",
      ],
      default: "JSON",
```

Then you can update the prop description:
>Format of the response message. If unspecified, `JSON` will be used for the response.